core/src/main/java/com/bankbazaar/core/manager/TripActivityEntityManager.java

```diff
@@ -0,0 +1,97 @@
+package com.bankbazaar.core.manager;
+
+import com.bankbazaar.core.model.TripActivityCompositeKey;
+import com.bankbazaar.core.model.TripActivityEntity;
+import com.bankbazaar.core.model.TripEntity;
+import com.bankbazaar.core.repository.TripActivityRepository;
+import com.bankbazaar.core.repository.TripRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+@Slf4j
+@Service
+public class TripActivityEntityManager {
+    @Autowired
+    private TripActivityRepository tripActivityRepository;
+
+    @Autowired
+    private TripRepository tripRepository;
+
+    /**
+     * Insert to TripActivity table
+     * @param data
+     */
+    public TripActivityEntity saveTripActivity(TripActivityEntity data)
+    {
+        Optional<TripEntity> tripId = tripRepository.findById(data.getTripId().getTripId());
```

Comment:
Actually, when i try to save "data", i get an error because all other fields in "data.tripId" is null. So, what i am doing is getting the existing "trip" from DB with "data.tripId.tripId" and then replacing the object "data.tripId" with the existing trip. This problem started when i set columns to "nullable=false".

Reply:
But when you are inserting a TripActivity why should it try to update the Trip table? Something does not seem right there

Reply:
I tried after removing that part. It is working now. I think previously there was something else that was causing the error.

Reply:
I have found the problem. It was a problem caused by the entity class because of unproper foreign key mapping. I have made the necessary changes.